src/platforms/modrinth_handler.py

```diff
@@ -0,0 +1,358 @@
+        plugin_download_name = f"{plugin_name}-{version}.jar"
+    
+    download_plugin_path = Path(f"{download_path}/{plugin_download_name}")
+    
+    # Use existing download infrastructure but with Modrinth URL
+    _download_modrinth_file(download_url, download_plugin_path)
+    
+    return None
+
+
+def _download_modrinth_file(url: str, download_path: Path) -> None:
+    """
+    Downloads a file from Modrinth - similar to GitHub download function
+    
+    :param url: Modrinth file download URL
+    :param download_path: Path where to save the file
+    :returns: None
+    """
+    import os
+    import requests
+    from zipfile import ZipFile
+    from rich.console import Console
+    from rich.progress import Progress
+    from src.utils.utilities import convert_file_size_down, remove_temp_plugin_folder
+    from src.handlers.handle_sftp import sftp_create_connection, sftp_upload_file
+    from src.handlers.handle_ftp import ftp_create_connection, ftp_upload_file
+    
+    config_values = config_value()
+    
+    # Use rich Progress() to create progress bar (same as existing implementations)
+    with Progress(transient=True) as progress:
+        header = {'user-agent': 'pluGET/1.0'}
+        r = requests.get(url, headers=header, stream=True)
+        try:
+            file_size = int(r.headers.get('content-length'))
+            # create progress bar
+            download_task = progress.add_task("    [cyan]Downloading...", total=file_size)
+        except TypeError:
+            # Content-length returned nothing
+            file_size = 0
+        with open(download_path, 'wb') as f:
+            # split downloaded data in chunks of 32768
+            for data in r.iter_content(chunk_size=32768):
+                f.write(data)
+                # don't show progress bar if no content-length was returned
+                if file_size == 0:
+                    continue
+                progress.update(download_task, advance=len(data))
+
+    # use rich console for nice colors (consistent with existing style)
+    console = Console()
+    if file_size == 0:
+        console.print(
+            f"    [not bold][bright_green]Downloaded[bright_magenta]         file [cyan]→ [white]{download_path}"
+        )
+    elif file_size >= 1000000:
+        file_size_data = convert_file_size_down(convert_file_size_down(file_size))
+        console.print("    [not bold][bright_green]Downloaded[bright_magenta] " + (str(file_size_data)).rjust(9) + \
+             f" MB [cyan]→ [white]{download_path}")
+    else:
+        file_size_data = convert_file_size_down(file_size)
+        console.print("    [not bold][bright_green]Downloaded[bright_magenta] " + (str(file_size_data)).rjust(9) + \
+             f" KB [cyan]→ [white]{download_path}")
+
+    # check if plugin file is a proper .jar-file (same validation as existing code)
+    try:
+        with ZipFile(download_path, "r") as plugin_jar:
+            plugin_jar.open("plugin.yml", "r")
```

Comment:
When testing with Veinminer from modrinth this will cause pluGET to say that the plugin was not correctly downloaded because the `plugin.yml` file was not found. Veinminer has only a `paper-plugin.yml` file in their root plugin archive. 
We test it the same way in the `plugin_downloader.py`. This needs to be fixed across pluGET so it will check if one of the files is present.  
I will create an issue for that. You can change it or leave it as this mirrors currently the same behaviour across all platforms.